policy_agent.py

```diff
@@ -106,6 +112,15 @@ def get_handler(self, resource_type, event_type):
         _log.debug("Looking up handler for event: %s", key)
         return self._handlers[key]
 
+    def load_cache(self):
+        """
+        Loads cache from etcd.
+        """
+        endpoints = self._client.get_endpoints()
```

Comment:
(Actually it's probably fine, there's a second filter step, although it won't scale very well because we pull every host from etcd.)